### PR TITLE
Fix event model query defaults

### PR DESCRIPTION
**Pull request by tn3rb**
## Problem this Pull Request solves
same issue as described in #523 but with regards to the event model

## How has this been tested
unit tests

### CODE REVIEW

File: assets/src/data/model/event/index.js [hunk 1]
```diff
@@ -87,29 +87,36 @@ export const mapOrderBy = ( orderBy ) => {
...
+} ) => {
 	const where = [];
 	const GREATER_AND_EQUAL = encodeURIComponent( '>=' );
 	const LESS_AND_EQUAL = encodeURIComponent( '<=' );
 
 	if ( ! showExpired ) {
-		where.push( 'where[Datetime.DTT_EVT_end**expired][]=>&where[Datetime.DTT_EVT_end**expired][]=' +
-			nowDateAndTime.local().format() );
+		where.push(
+			'where[Datetime.DTT_EVT_end**expired][]=>' +
```
Comment (nerrad) [inline assets/src/data/model/event/index.js]: huh, I think I've missed this in multiple places (i.e. will affect our datetime and ticket implementation as well).  This line results in (with the other two):

`where[Datetime.DTT_EVT_end**expired][]=>&where[Datetime.DTT_EVT_end**expired[]='{some-date-string}'`

It doesn't makes sense?  I think the first part which is `where[Datetime.DTT_EVT_end**expired][]=>&` shouldn't even be present right?

Comment (nerrad) [reply assets/src/data/model/event/index.js]: actually no, it's correct.  But we should be using `GREATER_AND_EQUAL` instead of `=>` in that line.

File: assets/src/data/model/event/test/index.js [hunk 2]
```diff
@@ -24,7 +26,10 @@ describe( 'mapOrderBy()', () => {
 
 describe( 'whereConditions()', () => {
 	it( 'returns expected default for empty object passed in', () => {
-		expect( whereConditions( {} ) ).toEqual( '' );
+		expect( whereConditions( {} ) ).toEqual(
+			'where[Datetime.DTT_EVT_end**expired][]=>' +
```
Comment (nerrad) [inline assets/src/data/model/event/test/index.js]: This will need fixed to be `GREATER_AND_EQUAL` instead of `=>` if the same fix is done in the method being tested.

File: assets/src/data/model/event/test/index.js [hunk 3]
```diff
@@ -46,17 +51,12 @@ describe( 'whereConditions()', () => {
 		};
 		expect( whereConditions( testObject ) ).toEqual(
 			'where[Datetime.DTT_EVT_end**expired][]=>' +
```
Comment (nerrad) [inline assets/src/data/model/event/test/index.js]: this will need changed from `=>` to `GREATER_AND_EQUAL` as well.

File: assets/src/data/model/event/test/index.js [hunk 4]
```diff
@@ -66,8 +66,7 @@ describe( 'getQueryString', () => {
 		expect( getQueryString() ).toEqual(
 			'limit=100&order=DESC&order_by=Datetime.DTT_EVT_start' +
 			'&where[Datetime.DTT_EVT_end**expired][]=>' +
```
Comment (nerrad) [inline assets/src/data/model/event/test/index.js]: This will need changed from `=>` to `GREATER_AND_EQUAL` as well.

File: assets/src/data/model/event/index.js (hunk 1 above)
Comment (tn3rb) [reply assets/src/data/model/event/index.js]: ya I never grokked that, but assumed it was something funny with the way the REST API worked

so... the correct line should be:

```
'where[Datetime.DTT_EVT_end**expired][]=' + GREATER_AND_EQUAL  +
```

or

```
'where[Datetime.DTT_EVT_end**expired][]' + GREATER_AND_EQUAL  +
```

I would think the former would be correct ( ie: replace `=>` with `=' + GREATER_AND_EQUAL` ),
but it sounds like you are saying the latter  ( ie: replace `=>` with GREATER_AND_EQUAL` ),
which makes no sense at all to me.

Comment (nerrad) [reply assets/src/data/model/event/index.js]: OH, actually maybe its supposed to be `=` + `>` only.  In which case it'd probably be good to add `GREATER_THAN` and `LESS_THAN` constants that encode the equal and less than characters.  Since multiple models will likely be using these constants, it'd probably also be good to move them into the `data/model/base.js` module and they can just be included where needed?

Comment (tn3rb) [reply assets/src/data/model/event/index.js]: kk no THAT makes sense and I totally agree.
will do that right now

**Comment by tn3rb** [comment]
will add documentation after #522 is merged because of structural changes to docs

File: assets/src/data/model/datetime/index.js [hunk 5]
```diff
@@ -89,22 +91,24 @@ export const whereConditions = ( {
...
+			'where[DTT_EVT_end**expired][]=' + GREATER_THAN +
+			'&where[DTT_EVT_end**expired][]=' +
+			nowDateAndTime.local().format()
+		);
 	}
 	if ( month && month !== 'none' ) {
-		where.push( 'where[DTT_EVT_start][]=' +
-			GREATER_AND_EQUAL +
+		where.push(
+			'where[DTT_EVT_start][]=' + GREATER_THAN +
```
Comment (nerrad) [inline assets/src/data/model/datetime/index.js]: Why did you change this from `GREATER_AND_EQUAL` to `GREATER_THAN`?  I think it should remain as it was as we're wanting to _include_ the start of the month right?

File: assets/src/data/model/datetime/index.js [hunk 6]
```diff
@@ -89,22 +91,24 @@ export const whereConditions = ( {
...
 			'&where[DTT_EVT_start][]=' +
-			moment().month( month ).startOf( 'month' ).local().format() );
-		where.push( 'where[DTT_EVT_end][]=' +
-			LESS_AND_EQUAL +
+			moment().month( month ).startOf( 'month' ).local().format()
+		);
+		where.push(
+			'where[DTT_EVT_end][]=' + LESS_THAN +
 			'&where[DTT_EVT_end][]=' +
```
Comment (nerrad) [inline assets/src/data/model/datetime/index.js]: I think this should remain `LESS_AND_EQUAL` here because we want to include the last day of the month in the query right?

File: assets/src/data/model/datetime/test/index.js [hunk 7]
```diff
@@ -40,19 +41,17 @@ describe( 'whereConditions()', () => {
...
-			'where[DTT_EVT_end**expired][]=>' +
+			'where[DTT_EVT_end**expired][]=' + GREATER_THAN +
 			'&where[DTT_EVT_end**expired][]=' + expectedNow +
-			'&where[DTT_EVT_start][]=' + GREATER_AND_EQUAL +
+			'&where[DTT_EVT_start][]=' + GREATER_THAN +
 			'&where[DTT_EVT_start][]=' + expectedStartofDate +
-			'&where[DTT_EVT_end][]=' + LESS_AND_EQUAL +
+			'&where[DTT_EVT_end][]=' + LESS_THAN +
 			'&where[DTT_EVT_end][]=' + expectedEndofDate +
 			'&where[Event.EVT_ID]=' + 20
```
Comment (nerrad) [inline assets/src/data/model/datetime/test/index.js]: These tests will need modified if my other comments are acted on.

File: assets/src/data/model/event/index.js [hunk 8]
```diff
@@ -87,31 +89,40 @@ export const mapOrderBy = ( orderBy ) => {
...
-		where.push( 'where[Term_Relationship.Term_Taxonomy.Term.slug]=' + categorySlug );
+		where.push(
+			'where[Term_Relationship.Term_Taxonomy.Term.slug]=' + categorySlug
+		);
 	}
 	if ( month && month !== 'none' ) {
-		where.push( 'where[Datetime.DTT_EVT_start][]=' +
-			GREATER_AND_EQUAL +
+		where.push(
+			'where[Datetime.DTT_EVT_start][]=' + GREATER_THAN +
```
Comment (nerrad) [inline assets/src/data/model/event/index.js]: As noted elsewhere, this should remain as `GREATER_AND_EQUAL` and not be changed.

File: assets/src/data/model/event/index.js [hunk 9]
```diff
@@ -87,31 +89,40 @@ export const mapOrderBy = ( orderBy ) => {
...
 			'&where[Datetime.DTT_EVT_start][]=' +
-			moment().month( month ).startOf( 'month' ).local().format() );
-		where.push( 'where[Datetime.DTT_EVT_end][]=' +
-			LESS_AND_EQUAL +
+			moment().month( month ).startOf( 'month' ).local().format()
+		);
+		where.push(
+			'where[Datetime.DTT_EVT_end][]=' + LESS_THAN +
```
Comment (nerrad) [inline assets/src/data/model/event/index.js]: As noted elsewhere, this should remain `LESS_AND_EQUAL` and not be changed.

File: assets/src/data/model/event/test/index.js [hunk 10]
```diff
@@ -37,26 +44,19 @@ describe( 'whereConditions()', () => {
...
-			'&where[Datetime.DTT_EVT_start][]=' +
-			expectedStartofDate +
-			'&where[Datetime.DTT_EVT_end][]=' +
-			LESS_AND_EQUAL +
-			'&where[Datetime.DTT_EVT_end][]=' +
-			expectedEndofDate,
+			'&where[Datetime.DTT_EVT_start][]=' + GREATER_THAN +
+			'&where[Datetime.DTT_EVT_start][]=' + expectedStartofDate +
+			'&where[Datetime.DTT_EVT_end][]=' + LESS_THAN +
+			'&where[Datetime.DTT_EVT_end][]=' + expectedEndofDate,
```
Comment (nerrad) [inline assets/src/data/model/event/test/index.js]: These will need modified if my earlier comments are acted on.

File: assets/src/data/model/ticket/index.js [hunk 11]
```diff
@@ -91,24 +93,21 @@ export const whereConditions = ( {
...
 		where.push(
-			'where[TKT_start_date][]=' + GREATER_AND_EQUAL +
+			'where[TKT_start_date][]=' + GREATER_THAN +
 			'&where[TKT_start_date][]=' +
 			moment().month( month ).startOf( 'month' ).local().format()
 		);
 		where.push(
-			'where[TKT_end_date][]=' + LESS_AND_EQUAL +
+			'where[TKT_end_date][]=' + LESS_THAN +
 			'&where[TKT_end_date][]=' +
```
Comment (nerrad) [inline assets/src/data/model/ticket/index.js]: as noted earlier, you need to retain `GREATER_AND_EQUAL` and `LESS_AND_EQUAL` instead of changing those.  The only thing needing implemented is the `GREATER_THAN` to replace the `>` in the `showExpired` condition.  For months, we must include the starting date and the ending date in the query otherwise a query for the month of June would exclude events falling on June 1 or June 30.

File: assets/src/data/model/ticket/test/index.js [hunk 12]
```diff
@@ -39,20 +40,18 @@ describe( 'whereConditions()', () => {
...
 		expect( whereConditions( testObject ) ).toEqual(
-			'where[TKT_end_date**expired][]=>' +
+			'where[TKT_end_date**expired][]=' + GREATER_THAN +
 			'&where[TKT_end_date**expired][]=' + expectedNow +
-			'&where[TKT_start_date][]=' + GREATER_AND_EQUAL +
+			'&where[TKT_start_date][]=' + GREATER_THAN +
 			'&where[TKT_start_date][]=' + expectedStartofDate +
-			'&where[TKT_end_date][]=' + LESS_AND_EQUAL +
+			'&where[TKT_end_date][]=' + LESS_THAN +
 			'&where[TKT_end_date][]=' + expectedEndofDate +
```
Comment (nerrad) [inline assets/src/data/model/ticket/test/index.js]: this will need adjusted if my earlier comments are acted on.
